### PR TITLE
[IMPROVE] Add setting to control Also send to channel checkbox in thread

### LINKED ISSUE #20695: Hotchillee - 'Also send to channel' Default On issue.
Context
Hotchillee is looking to move its Digital cycling club communications from Slack to a Rocketchat based app. In doing so it hopes to maintain the user experience as much as possible so as not to confuse its user base while using Rocketchats white labelling capability to give it Hotchillee brand and feel. As part of this Hotchillee looks to retain all the historic chat from Slack to make it as seamless as possible for users transferring to the new platform (similar to when Facebook does a major UX overhaul - all the data remains but the look and feel is different and new features are added but it’s still familiar enough not to alienate users).

### Description:

When writing in threads there is a checkbox for ‘Also send to channel’. As admins we have selected to have this unchecked by default. In the app this works as expected but on desktop, when writing the first message in a thread this is checked by default. When someone goes up to write the second or subsequent message in those threads it is turned off as expected. It is just in the browser that this issue occurs.

### Steps to reproduce:

1. Visit our clean test instance https://community.hotchillee.cc/channel/general_chat and click on any thread “reply” button to see for yourselves
2. Write a reply
3. Click on the same thread again to reply
4. See the checkbox has now defaulted to off

### Expected behavior:

The 'Also send to Channel' checkbox should be defaulted to Off at all times

### Actual behavior:

The 'Also send to Channel' button is defaulted to 'On' for the first thread reply (see video).

### Server Setup Information:

- Version of Rocket.Chat Server: 3.11.0
- Operating System: 
- Deployment Method: Docker on Kubernetes via Helm
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: 12.18.4
- MongoDB Version: 4.2.12
- App builder version: 1.22.1

### Client Setup Information

- Desktop App or Browser Version: Chrome and Safari - Latest versions
- Operating System: Mac OS

### Additional context

Not applicable

### Relevant logs:

Not applicable.

https://user-images.githubusercontent.com/78903354/107631382-c8684600-6c5c-11eb-8719-ef52591f693a.mov

### PR DESCRIPTION
**Pull request by Darshilp326**
## Proposed changes (including videos or screenshots)

image

With the setting **turned on**, the checkbox will maintain the behavior, Also sending to the channel checked on the **first message**
With the setting **turned off**, the checkbox won't be checked by default on the first message.

https://user-images.githubusercontent.com/55157259/110026378-8f505c80-7d56-11eb-8350-7fbdfdd6f570.mp4

## Issue(s)

Closes #20695

### CODE REVIEW

**Review by ggazzo — DISMISSED** [review]
Please create setting under `admin -> threads` and let the old behavior enabled by default

**Comment by Darshilp326** [comment]
@ggazzo By old behaviour do you mean to send to the channel by default?

**Comment by dougfabris** [comment]
> @ggazzo By old behaviour do you mean to send to the channel by default?

@Darshilp326 yeah! he wanted to mean to stay 'send to the channel' by default and create an option on  `admin -> threads` to have a behavior like you did if the admin wants to. Can you help us with that?

**Comment by Darshilp326** [comment]
@dougfabris Yes, I would love to work on this issue.

**Comment by Darshilp326** [comment]
@dougfabris Please review :smile:

**Comment by dougfabris** [comment]
> @dougfabris Please review 😄

@Darshilp326 It's not what we need 😟
The setting you used is for another thing. What we need is a new setting on `admin -> threads` that add control to "Also send to channel". This new setting has to be checked by default so we'll get always the checkbox checked. If the admin wants to he can uncheck and then every new thread will come with the checkbox unchecked. Let me know if you're interested to continue improving this fix and feel free to contact me for more details! 
Thanks for helping us! 😁

**Comment by Darshilp326** [comment]
> > @dougfabris Please review smile
> 
> @Darshilp326 It's not what we need. The setting you used is for another thing 
> What we need is a new setting on `admin -> threads` that add control to "Also send to channel". This new setting has to be checked by default so we'll get always the checkbox checked. If the admin wants to he can uncheck and then every new thread will come with the option unchecked.
> 
> Let me know if you're interested to continue improving this fix and feel free to contact me for more details!
> Thanks for helping us!

Ohk got your point. Working on it! :sweat_smile:

**Comment by Darshilp326** [comment]
https://user-images.githubusercontent.com/55157259/110026378-8f505c80-7d56-11eb-8350-7fbdfdd6f570.mp4

**Comment by Darshilp326** [comment]
@dougfabris Had made the requested changes! :smiley:

**Review by dougfabris — APPROVED** [review]

**Review by zdumitru — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]

**Review by dougfabris — APPROVED** [review]

File: apps/meteor/app/threads/server/settings.ts [hunk 1]
```diff
@@ -7,4 +7,9 @@ settingsRegistry.addGroup('Threads', function () {
 		type: 'boolean',
 		public: true,
 	});
+	this.add('Also_send_to_channel_default', true, {
```
Comment (ggazzo) [inline apps/meteor/app/threads/server/settings.ts:10]:
```suggestion
	this.add('Also_send_to_channel_by_default', true, {
```

**Review by ggazzo — CHANGES_REQUESTED** [review]

**Comment by Weterok** [comment]
really need this feature

**Comment by dougfabris** [comment]
> How long for wait for a review? This is a very necessary feature and it's frozen...

@zhq745w We're planning to have this on the next major release, this month!

**Comment by dougfabris** [comment]
Hey @Darshilp326
After a long time discussing internally we decided to move forward with the solution here #26002, thanks a lot for your help!

**Comment by ulope** [comment]
How is #26002 a solution? Removing the preference does not solve the issue on an organisation wide level.

**Comment by dougfabris** [comment]
@ulope As I described in the breaking change message: "In the past, we used the `show message in main thread` as a solution to help users to understand the thread feature, as this feature is now mature enough there's no reason to maintain this preference." which causes confusion in a lot of users and customers. Additionally, we introduced a new preference allowing users to control the behavior of Also send to channel checkbox, so you users can choose or not to send the messages in the main message

**Pull request closed without merging**